cunumeric/module.py

```diff
@@ -1524,17 +1532,19 @@ def dstack(tup):
     --------
     GPU, CPU
     """
-    fname = dstack.__name__
-    tup, common_info = check_shape_dtype(tup, fname)
     # Reshape arrays to (1,N,1) for ndim ==1 or (M,N,1) for ndim == 2:
-    if common_info.ndim <= 2:
-        shape = list(tup[0].shape)
-        if common_info.ndim == 1:
-            shape.insert(0, 1)
-        shape.append(1)
-        common_info.shape = shape
-        for i, arr in enumerate(tup):
-            tup[i] = arr.reshape(shape)
+    reshaped = []
+    inputs = list(ndarray.convert_to_cunumeric_ndarray(inp) for inp in tup)
+    for arr in inputs:
+        if arr.ndim <= 2:
```

Comment:
A suggestion:
```
if arr.ndim == 1:
  arr = arr.reshape((1,) + arr.shape + (1,))
elif arr.ndim == 2:
  arr = arr.reshape(arr.shape + (1,))
```
Nested conditions in this if block seem to make the code unnecessarily obscure.